include/libcyphal/transport/can/can_transport.hpp

```diff
@@ -6,9 +6,9 @@
 #ifndef LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
 #define LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
 
-#include "libcyphal/transport/can/media.hpp"
 #include "libcyphal/transport/errors.hpp"
 #include "libcyphal/transport/transport.hpp"
+#include "media.hpp"
```

Comment:
B/c this "island" of `"..."` includes are **relative** ones:
- either relative from **current** file location (this is why just `"media.hpp"` - it is a sibling)
- or from root of includes (aka `<repo>/include/`)

Note that instead of f.e. `#include "libcyphal/transport/errors.hpp"` another option will be `#include "../errors.hpp"`, but I decided against usage of such `..` (grand-)parent folder references. But in case of sibling/children includes IMO it makes sense use shorter version - f.e. if I would need rename ".../can/..." part of the path then it will zero changes for those files inside of the "can" folder.

Reply:
I do not believe that the typical behaviour of looking first in the current folder is actually controlled by the c++ standard.

Reply:
@serges147 This is true. The standard is incredibly vague on the specifics of the search behavior.

![image](https://github.com/OpenCyphal-Garage/libcyphal/assets/3298404/d09fc849-e79e-4126-9bed-1c1f85b5a1d9)

Reply:
ok, but I believe that a lot of existing c++ codebases will stop to compile if `xxx.hpp` won't be searched for `#include "xxx.hpp"` at sibling `xxx.cpp` anymore...

Also, here is what I used as a guideline: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-incform
Although, if I followed this guideline strict then I would use more `<>`... Instead... but I think found middle ground, namely:
- use `""` syntax for your own "local" stuff
- use `<>` for anything which is external, like: CETL, lizards, std, or **even** libcyphal includes themselves if they are referenced in unit tests (I treat libcyphal's unit tests as "external" client/user of the headers).

Also please note everywhere at libcyphal (and it's unit tests) this approach is applied consistently, so it's not like here I use one approach and in other place different one - I indeed care about inclusions of headers very much, grouping them to separate logical islands, keeping their order in very strict specific order, and eliminating anything that became unused.

Anyway, please let me know if you guys decide that, still, we should not rely on "search first local/relative" behavior, but instead ALWAYS use "fullpath" `#include "libcyphal/xxx/yyy.hpp"` regardless from where it was included at a `libcyphal/...` source file). I will apply (consistently of course) such changes everywhere. Thanks!

Reply:
I wasn't aware of that core guidelines rule. The guidelines are not the standard but as they are published by Stroustrup and Sutter, we can probably rely on them here?